sktools/sk

```diff
@@ -0,0 +1,214 @@
+    >&2 echo "*********************************************"
+    exit 1
+fi
+
+if [ $command == "init" ]; then
+    echo "{" > "$project"
+    echo "  \"skipVersion\": \"1.0\"," >> "$project"
+    echo "  \"programUnits\": {" >> "$project"
+    echo "    \"a.out\": {" >> "$project"
+    echo "      \"kind\": \"Program\"," >> "$project"
+    echo "      \"sources\": [\".\"]," >> "$project"
+    echo "      \"references\": [{ \"path\": \"$prelude\" }]" >> "$project"
+    echo "    }" >> "$project"
+    echo "  }" >> "$project"
+    echo "}" >> "$project"
+    exit 0
+fi
+
+if [ $command == "run" ]; then
+    if ! [ -f "$skdir/bin.out" ]; then
+	>&2 echo "*********************************************"
+	>&2 echo "ERROR: could not find any binary file!"
+	>&2 echo "*********************************************"
+	>&2 echo "Did you run?: $ sk build \"$root\""
+	exit 8
+    fi
+    exec "$skdir/bin.out" "${@:2}"
+fi
+
+if ! [ -f "$project" ]; then
+    >&2 echo "*********************************************"
+    >&2 echo "ERROR: $project doesn't exist!"
+    >&2 echo "*********************************************"
+    >&2 echo "You can create a default project file with:"
+    >&2 echo "\$ sk init $root"
+    exit 7
+fi
+
+server_pid=`ps xa | grep skserver | grep "$root" | awk '{ print $1 }'`
+if [ $command == "stop" ]; then
+    if [ -z "$server_pid" ]; then
+	>&2 echo "Error: could not stop server, no server found"
+	exit 3
+    else
+	kill $server_pid
+	exit 0
+    fi
+fi
+
+if [ $command == "start" ]; then
+    if [ -f "$skdir/lock" ]; then
+	>&2 echo "*********************************************"
+	>&2 echo "Error: cannot start server."
+	>&2 echo "Another server is already running."
+	>&2 echo "*********************************************"
+	>&2 echo "If you don't think that should be the case run:"
+        >&2 echo "\$ rm -Rf \"$skdir\""
+	exit 4
+    else
+	"$skserver" "$root" &
+	echo "Server started for: $root"
+	exit 0
+    fi
+fi
+
+if ! [ $command == "start" ]; then
+
+    # The skserver is not running.
+    if ! [ -f "$skdir/lock" ]; then
+	>&2 echo "*********************************************"
+	>&2 echo "No skserver is running for $root"
+	>&2 echo "Run: sk start $root"
+	>&2 echo "*********************************************"
+	exit 5
+    fi
+
+    # The skserver is running, but skip_server died
+    skip_server_pid=`ps xa | grep skip_server | grep "$root" | awk '{ print $1 }'`
+    if [ -z "$skip_server_pid" ]; then
+	>&2 echo "Skip_server died unexpectedly.";
+	>&2 echo "Re-run:"
+	>&2 echo "\$ sk start $root";
+	if [ "$server_pid" ]; then
+	    kill $server_pid
+	fi;
+	exit 6
+    fi
+fi
+
+if [ $command == "status" ]; then
+    if [ -f "$skdir/lock" ]; then
+	echo "Skip server is running"
+    else
+	echo "No server running"
+    fi;
+    exit 0
+fi
+
+rm -f "$errfile"
+rm -f "$skdir/out.ll"
+rm -f "$skdir/preamble_and_out.ll"
+rm -f "$skdir/out.o"
+rm -f "$skdir/bin.out"
+
+read < "$skdir/fifo_response" &
```

Comment:
It's weird that here we read/write `$skdir/fifo_*`, yet later in the file we declare the `fifo_*` variables to abstract away the details.  Refactor to make this less brittle?